starport/pkg/cosmosfaucet/transfer.go

```diff
@@ -55,9 +55,14 @@ func (f Faucet) Transfer(ctx context.Context, toAccountAddress string, amount ui
 		return err
 	}
 
-	if f.coinsMax[denom] != 0 && totalSent >= f.coinsMax[denom] {
-		return fmt.Errorf("account has reached maximum credit allowed per account (%d)", f.coinsMax[denom])
+	if (f.coinsMax[denom] != 0) {
+		if (totalSent >= f.coinsMax[denom]) {
+			return fmt.Errorf("account has reached maximum credit allowed per account (%d)", f.coinsMax[denom])
+		}else if ((totalSent+amount) >= f.coinsMax[denom]) {
```

Comment:
```suggestion
		} else if (totalSent+amount) >= f.coinsMax[denom] {
```

Reply:
Isn't this check sufficient? If we do this `totalSent` will never be higher than `f.coinsMax[denom]`?

Reply:
Nope...same mistake as the original one.

if totalSent< coinsMax the first check would pass allowing the faucet to send 1 gazillion amount :P . Need to ensure that totalSent (in the past) + amount requested (now)  does not exceed coinsMax